src/ethereum/EthereumNetwork.ts

```diff
@@ -1656,13 +1755,30 @@ export class EthereumNetwork {
         currencyCodes.push(currencyCode)
       }
 
-      for (const tk of currencyCodes) {
+      // If this engine supports the batch token balance query, no need to check
+      // each currencyCode individually.
+      const { ethBalCheckerContract } = this.ethEngine.networkInfo
+
+      if (ethBalCheckerContract != null) {
         await this.checkAndUpdate(
-          this.ethNeeds.tokenBalLastChecked[tk],
+          this.ethNeeds.tokenBalsLastChecked,
           BAL_POLL_MILLISECONDS,
           preUpdateBlockHeight,
-          async () => await this.check('tokenBal', tk)
+          async () => await this.check('tokenBal')
         )
+      }
+
+      for (const tk of currencyCodes) {
+        // Only check each code individually if this engine does not support
+        // batch token balance queries.
+        if (ethBalCheckerContract == null) {
```

Comment:
If you move this for loop block into an else condition above you can remove this duplicate ethBalCheckerContract check